daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/SimpleTypes.scala

```diff
@@ -312,40 +308,49 @@ abstract class SimpleTypeDefBase(xml: Node, lexicalParent: SchemaComponent)
     }
   }
 
-  lazy val optInputTypeCalc = findPropertyOption("inputTypeCalc", expressionAllowed = true)
-  lazy val optOutputTypeCalc = findPropertyOption("outputTypeCalc", expressionAllowed = true)
-
   lazy val optTypeCalculator: Option[TypeCalculator] = LV('optTypeCalculator) {
```

Comment:
Will do, I had to learn about type calculators too so this would definitely be useful. Quickly, type calculators are used to convert a parsed value (based on the repType) to a value that actually ends up in the infoset.  There are different type calculators for supporting conversions based on enumerations and unions, and previously expressions, which is now removed.

So a parser actually parses the data (based on the repType), and the type calculator converts that parsed value to something else (based on the enumeration/union), usually with a different type, and that ends up in the infoset.